pkg/storage/storage.go

```diff
@@ -104,48 +127,52 @@ type RelationshipTupleReader interface {
 	Write(ctx context.Context, store string, d Deletes, w Writes) error
 
-	// MaxTuplesPerWrite returns the maximum number of items (writes and deletes combined) allowed in a single write transaction
+	// MaxTuplesPerWrite returns the maximum number of items (writes and deletes combined)
+	// allowed in a single write transaction.
 	MaxTuplesPerWrite() int
 }
 
-// ReadStartingWithUserFilter specifies the filter options that will be used to constrain the ReadStartingWithUser
-// query.
+// ReadStartingWithUserFilter specifies the filter options that will be used
+// to constrain the [RelationshipTupleReader.ReadStartingWithUser] query.
 type ReadStartingWithUserFilter struct {
 	ObjectType string
 	Relation   string
 	UserFilter []*openfgav1.ObjectRelation
 }
 
+// ReadUsersetTuplesFilter specifies the filter options that
+// will be used to constrain the UserType query.
 type ReadUsersetTuplesFilter struct {
-	Object                      string                         // required
-	Relation                    string                         // required
-	AllowedUserTypeRestrictions []*openfgav1.RelationReference // optional
+	Object                      string                         // Required.
+	Relation                    string                         // Required.
+	AllowedUserTypeRestrictions []*openfgav1.RelationReference // Optional.
 }
 
-// AuthorizationModelReadBackend Provides a Read interface for managing type definitions.
+// AuthorizationModelReadBackend provides a read interface for managing type definitions.
 type AuthorizationModelReadBackend interface {
-	// ReadAuthorizationModel Read the model corresponding to store and model id
-	// If it's not found, it must return ErrNotFound
+	// ReadAuthorizationModel reads the model corresponding to store and model ID.
+	// If it's not found, it must return ErrNotFound.
 	ReadAuthorizationModel(ctx context.Context, store string, id string) (*openfgav1.AuthorizationModel, error)
 
-	// ReadAuthorizationModels Read all type definitions ids for the supplied store.
+	// ReadAuthorizationModels reads all type definitions ids for the supplied store.
 	ReadAuthorizationModels(ctx context.Context, store string, options PaginationOptions) ([]*openfgav1.AuthorizationModel, []byte, error)
 
-	// FindLatestAuthorizationModelID Returns the last model `id` written for a store.
-	// If none were ever written, it must return ErrNotFound
+	// FindLatestAuthorizationModelID returns the last model `id` written for a store.
+	// If none were ever written, it must return ErrNotFound.
 	FindLatestAuthorizationModelID(ctx context.Context, store string) (string, error)
 }
 
-// TypeDefinitionWriteBackend Provides a write interface for managing typed definition.
+// TypeDefinitionWriteBackend provides a write interface for managing typed definition.
 type TypeDefinitionWriteBackend interface {
```

Comment:
In a different PR this could be renamed to `AuthorizationModelWriteBackend` for more consistency

Reply:
Wouldn't that be a breaking change?

Reply:
I don't think so 🤔 to have their own custom datastore, people need to implement `OpenFGADatastore`, not this one

Reply:
Yes but this is a publicly exported type on a v1+ release, renaming this would technically be a breaking change, unless we have data that supports this is not used across public and private repositories anywhere?